### PR TITLE
Add Postman add-on

### PR DESCRIPTION
**Pull request by VitikaSoni**
Related to https://github.com/zaproxy/zaproxy/issues/6960

This is the first PR for my GSoC'23 project - [Import Postman API Definitions into OWASP ZAP](https://summerofcode.withgoogle.com/programs/2023/projects/OlBxaE5X). Please let me know if anyone has any questions/suggestions regarding the project :)

Signed-off-by: VitikaSoni <vitika.program@gmail.com>

### CODE REVIEW

File: addOns/postman/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+version=1
```
Comment (thc202) [inline addOns/postman/gradle.properties:1]: `0.0.1`

File: addOns/postman/postman.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+description = "Imports and spiders Postman definitions."
+
+zapAddOn {
+    addOnName.set("Postman Support")
+    zapVersion.set("2.12.0")
+
+    manifest {
+        author.set("ZAP Dev Team")
```
Comment (thc202) [inline addOns/postman/postman.gradle.kts:8]: Add `url.set("https://www.zaproxy.org/docs/desktop/addons/postman-support/")`

File: addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2014 The ZAP Development Team
```
Comment (thc202) [inline addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java:6]: `2023`

File: addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ */
+package org.zaproxy.addon.postman;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+
+public class ExtensionPostman extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionSimpleExample";
```
Comment (thc202) [inline addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java:28]: Needs update.

File: addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        super.hook(extensionHook);
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public String getDescription() {
```
Comment (thc202) [inline addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java:45]: Would be good to override `getUIName()` too.

File: addOns/postman/src/main/javahelp/org/zaproxy/addon/postman/resources/help/map.jhm [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!DOCTYPE map
+  PUBLIC "-//Sun Microsystems Inc.//DTD JavaHelp Map Version 1.0//EN"
+         "http://java.sun.com/products/javahelp/map_1_0.dtd">
+
+<map version="1.0">
+	<mapID target="postman-icon" url="contents/images/postman.png" />
+    <mapID target="postman" url="contents/postman.html" />
```
Comment (thc202) [inline addOns/postman/src/main/javahelp/org/zaproxy/addon/postman/resources/help/map.jhm:8]: Use the prefix `addon.*` for the targets.

**Comment by VitikaSoni** [comment]
I have made the suggested changes :)

File: addOns/postman/src/main/javahelp/org/zaproxy/addon/postman/resources/help/map.jhm (hunk 6 above)
Comment (thc202) [reply addOns/postman/src/main/javahelp/org/zaproxy/addon/postman/resources/help/map.jhm]: The other `target` references need to be updated to match the new value.

Comment (VitikaSoni) [reply addOns/postman/src/main/javahelp/org/zaproxy/addon/postman/resources/help/map.jhm]: @thc202 Done

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by ricekot — APPROVED** [review]
Thanks!

**Comment by kingthorin** [comment]
Whoop whoop, step 1 !!! 🎉

File: addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.postman;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+
+public class ExtensionPostman extends ExtensionAdaptor {
```
Comment (bharath-k-7) [inline addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java:26]: @VitikaSoni Does this import the requests with headers and body from the collection? In v2.14.0, I can import the requests but the headers & body is not imported

Comment (thc202) [reply addOns/postman/src/main/java/org/zaproxy/addon/postman/ExtensionPostman.java:26]: Please, use the ZAP User Group for usage questions: https://groups.google.com/group/zaproxy-users

**Comment by Himan10** [comment]
Hey guys. 
I have the same use case as others for the Postman API Collection, I am able to import the postman API collection but not able to perform scans on the APIs given in the postman collection. 

Could you please let me know how's that work? 
FYI, I am using docker image of zaproxy and mostly performing all the stuff with `.zap-api-scan.py` and `zap-full-scan.py`

**Comment by Himan10** [comment]
Hey @kingthorin 

I have the same use case as others for the Postman API Collection, I am able to import the postman API collection but not able to perform scans on the APIs given in the postman collection. 

Could you please let me know how's that work? 
FYI, I am using docker image of zaproxy and mostly performing all the stuff with `.zap-api-scan.py` and `zap-full-scan.py`

**Comment by thc202** [comment]
Please use the [User Group](https://groups.google.com/group/zaproxy-users) for usage questions.

**Pull request merged**